sdk/Lusid.Sdk.Tests/Utilities/ApiRetryHandlerTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Net;
+using System.Threading;
+using Lusid.Sdk.Api;
+using Lusid.Sdk.Client;
+using Lusid.Sdk.Utilities;
+using NUnit.Framework;
+using Polly;
+using RestSharp;
+
+namespace Lusid.Sdk.Tests.Utilities
+{
+    [TestFixture]
+    public class ApiRetryHandlerTests
+    {
+        private LusidApiFactory _apiFactory;
+        private readonly Policy<IRestResponse> _initialRetryPolicy = RetryConfiguration.RetryPolicy;
+        private HttpListener _httpListener;
+        private const string ListenerUriPrefix = "http://localhost:4444/";
+        
+        [SetUp]
+        public void SetUp()
+        {
+            _httpListener = new HttpListener();
+            _httpListener.Prefixes.Add(ListenerUriPrefix);
+            
+            var testApiConfig = TestLusidApiFactoryBuilder.CreateApiConfiguration(null);
+            testApiConfig.ApiUrl = ListenerUriPrefix;
+            
+            _apiFactory = new LusidApiFactory(testApiConfig);
+        }
+
+        [Test]
+        [Explicit("This passes locally, but not in the pipeline, and no idea why. Setting to explicit until resolved.")]
+        public void CallGetPortfoliosApi_WhenPollyRetryPolicyConfigured_PollyIsTriggered()
+        {
+            const int expectedNumberOfRetries = ApiRetryHandler.MaxRetryAttempts;
+            _httpListener.Start();
+            for (var i = 0; i < expectedNumberOfRetries + 1; i++)
+            {
+                 _httpListener.BeginGetContext(result =>
+                {
+                    var listener = (HttpListener) result.AsyncState;
+                    // Call EndGetContext to complete the asynchronous operation.
+                    var context = listener.EndGetContext(result);
+
+                    // Obtain a response object.
+                    var response = context.Response;
+
+                    // Abort the response. This returns 0 status code.
+                    response.Abort();
+                }, _httpListener);
+            }
+            
+            var currentRetry = 0;
+            var testRetryPolicy = Policy
+                .HandleResult<IRestResponse>(restResponse => restResponse.StatusCode == 0)
+                .Retry(
+                    retryCount: expectedNumberOfRetries,
+                    (result, i) =>
+                    {
+                        Console.WriteLine($"If you see this, then polly retry works. Retry number: {i}");
+                        currentRetry = i;
+                    });
+
+            RetryConfiguration.RetryPolicy = testRetryPolicy;
+
+            // Calling GetPortfolio or any other API triggers the flow that triggers polly
+            var sdkResponse = _apiFactory.Api<IPortfoliosApi>().GetPortfolio("any", "any");
+            
+            Assert.That(currentRetry, Is.EqualTo(expectedNumberOfRetries));
+            // In the future 0 error codes with throw an error after retries exceeded
+            Assert.That(sdkResponse, Is.Null);
```

Comment:
So if the maximum retries are reached the response is `null`? Why is that?

Reply:
because it retries 3 times, and still gets the same response as it got the first time around. Which was null. 

(When we implement the exception factory handler, this will then throw an exception after the 3rd retry)

Reply:
Okay.